src/util/get-github-client-config.ts

```diff
@@ -0,0 +1,89 @@
+// this will need to be updated later to check for database entry
+import { Installation } from "models/installation";
+import { GitHubServerApp } from "models/git-hub-server-app";
+import { GitHubInstallationClient } from "../github/client/github-installation-client";
+import { getCloudInstallationId } from "../github/client/installation-id";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+import { GitHubUserClient } from "../github/client/github-user-client";
+import Logger from "bunyan"
+import { GitHubAppClient } from "../github/client/github-app-client";
+
+export const GITHUB_CLOUD_HOSTNAME = "https://github.com";
+export const GITHUB_CLOUD_API_BASEURL = "https://api.github.com";
+export const GITHUB_CLOUD_ACCEPT_HEADER = "application/vnd.github.v3+json"
+
+export interface GitHubClientConfig {
+	hostname: string;
+	apiBaseUrl: string;
+	acceptHeader: string;
+}
+
+export async function getGitHubApiUrl(jiraHost: string) {
+	const gitHubClientConfig = await getGitHubClientConfig(jiraHost);
+
+	return await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost) && gitHubClientConfig
+		? `${gitHubClientConfig.apiBaseUrl}`
+		: GITHUB_CLOUD_API_BASEURL;
+}
+
+export async function getGitHubHostname(jiraHost: string) {
+	const gitHubClientConfig = await getGitHubClientConfig(jiraHost);
+
+	return await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost) && gitHubClientConfig
+		? gitHubClientConfig.hostname
+		: GITHUB_CLOUD_HOSTNAME;
+}
+
+const getGitHubClientConfig = async (jiraHost: string): Promise<GitHubClientConfig> => {
+	// TODO: remove duplicate jiraHosts and find issue that causes duplicates
```

Comment:
We have ARC-772 to remove duplicate entries from the db and ARC-769 to add unique constraints to jiraHost and clientKey in the Installations table. Both are assigned to the `DISTURBED` epic. We should probably get whoever is disturbed in upcoming weeks to tackle these 2 issues.

Reply:
Probably need to remove it form disturbed and instead put it as part of the GHE work instead